### ISSUE TITLE
[Feature]: Introduce Project Level Config File

### ISSUE THREAD

**Issue by jxtngx** [wandb/wandb#5213]
### Description

Introduce a project level config file that the W&B Api, Agent et al will use in place of requiring users to set environment variables per project.

### Suggested Solution

Introduce a project level .yml, .json, .ini, or .cfg that will allow users to set environment configs in that file.

### Alternatives

Provide guidance on how to use a .env or programmatically setting environment variables.

### Additional Context

As it is now, users are required to set environment variables per project. With regard to logs and config, this means that if a user forgets to set an environment variable, Sweep logs will be sent to one place, run logs to another, and a config file is placed in user's root directory under .config/wandb.

There are currently 29 possible [environment variables](https://docs.wandb.ai/guides/track/environment-variables#optional-environment-variables) listed in the docs a user may have to set per project, and possibly deconflict in a .zshrc or .bash_profile if previously set during other projects.

**Comment by nate-wandb** [comment]
Hi @JustinGoheen, thank you the feature request. Do you think it would make sense to add env variable support to our file [based config](https://docs.wandb.ai/guides/track/config#file-based-configs) workflow so these could be set there?

Thank you,
Nate

**Comment by JustinGoheen** [comment]
Yes! That would be perfect since it would consolidate to a single file.
